app/Http/Controllers/BarbersController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Barber;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Gate;
+
+class BarbersController extends Controller
+{
+    public function store(Request $request)
+    {
+        Barber::create($request->all());
+
+        return redirect()->route('barbershop.register');
+    }
+
+    public function show(Barber $barber)
+    {
+        return view('barber.profile')->with('barber', $barber);
+    }
+
+    public function update(Barber $barber, Request $request)
+    {
+        if (Gate::allows('alter-barber', $barber)) {
```

Comment:
I *think* you can use the gate middleware to avoid having to do this in every method?